### DISCUSSION TITLE
MinimalBox and Table

### DISCUSSION THREAD

**Discussion by douglott** [QuestPDF/QuestPDF#559] [Q&A] [answered]
Hi,

I am trying to figure out how to get a table to take a minimal amount of horizontal space within a row without using constant width columns.  The table does not seem to be respecting the MinimalBox rule.  Following is the layout and screenshot of the results.  Thanks in advance for any help.

`column.Item().Row(row =>
                {
                    row.RelativeItem().Background(Colors.Blue.Lighten3).MinimalBox().Table(table =>
                    {
                        table.ColumnsDefinition(columns =>
                        {
                            columns.RelativeColumn();
                            columns.RelativeColumn();
                        });
                        table.Cell().Element(HeaderCell).Text("Shipment #");
                        table.Cell().Element(RegularCell).Text(_model.ShipmentNumber);
                        table.Cell().Element(HeaderCell).Text("SO #");
                        table.Cell().Element(RegularCell).Text(_model.SoNumber);
                        table.Cell().Element(HeaderCell).Text("Customer");
                        table.Cell().Element(RegularCell).Text(_model.ProjectName);
                    });`

`
    static IContainer HeaderCell(IContainer container)
    {
        return container
            .Border(1)
            .BorderColor(Colors.Grey.Lighten1)
            .Background(Colors.Grey.Lighten3)
            .Padding(3);
    }
    
    static IContainer RegularCell(IContainer container)
    {
        return container
            .Border(1)
            .BorderColor(Colors.Grey.Lighten1)
            .Padding(3);
    }`

<img width="913" alt="Screenshot 2023-05-10 at 11 16 08 AM" src="https://github.com/QuestPDF/QuestPDF/assets/19208649/57bab527-7732-41b2-bbc1-2ca68873f67a">

**Comment by MarcinZiabek** [comment, marked as answer]
The `Table` element does not support scaling down in size to the smallest child element. This type of automated algorithm is usually unstable and less efficient, depending on the content. It is often used during web development, where you have screens and contents of various sizes. In QuestPDF, when dealing with well-defined page sizes, I decided to rely on a solid and configurable approach, which behavior is easier to predict.

That being said, you can try to use the `Row.AutoItem` API that does scale to the smallest child. Just don't forget to limit the maximum width, e.g.:

```csharp
.Row(row => 
{
    row.AutoItem().MaxWidth(200).Text("My content");
});
```

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
